api/services/uaaStrategy.js

```diff
@@ -33,4 +35,34 @@ function createUAAStrategy(options, verify) {
   return strategy;
 }
 
-module.exports = { createUAAStrategy };
+async function verifyUAAUser(accessToken, refreshToken, profile, uaaGroups) {
+  const { user_id: uaaId, email } = profile;
+
+  const client = new UAAClient(accessToken);
+  const isVerified = await client.verifyUserGroup(uaaId, uaaGroups);
+
+  if (!isVerified) {
+    return null;
+  }
+
+  const identity = await UAAIdentity.findOne({ where: { email } });
+
+  if (!identity) {
+    return null;
+  }
+
+  const user = await identity.getUser();
+
+  if (!user) {
+    return null;
+  }
+
+  identity.uaaId = uaaId;
+  identity.accessToken = accessToken;
+  identity.refreshToken = refreshToken;
+  await identity.save();
```

Comment:
meh, maybe preference, but I think you can: 
```
await identity.update({ uaaId, accessToken, refreshToken });
```

Reply:
Copy that.